src/main/java/org/casbin/jcasbin/main/CoreEnforcer.java

```diff
@@ -424,6 +417,17 @@ public boolean enforce(Object... rvals) {
 
         return result;
     }
+    
+    private void getRTokens(Map<String, Object> parameters, Object ...rvals) {
+      for(String rKey : model.model.get("r").keySet()) {
+        if(!(rvals.length == model.model.get("r").get(rKey).tokens.length)) { continue; }
+        for (int j = 0; j < model.model.get("r").get(rKey).tokens.length; j ++) {
+          String token = model.model.get("r").get(rKey).tokens[j];
+          parameters.put(token, rvals[j]);
+        }
+   
+      }
+    }    
```

Comment:
This code block access over all the provided request and selects with the one matching the actual request token length.
Earlier, it was selecting just one rule where this adds flexibility for more set of request rule.